### PR TITLE
Get audio and video mimes from GStreamer, fixes #3826

### PR DESCRIPTION
**Pull request by SAMdroid-apps**
Replaces #185 

GCI Ticket:  http://www.google-melange.com/gci/task/view/google/gci2014/5810064153640960

@godiard Please see the comment about application mimes, line 112

### CODE REVIEW

**Comment by quozl** [comment]
What is the impact on startup time?

**Comment by SAMdroid-apps** [comment]
Very little.  I didn't notice anything.

The 1st patch took about 0.6s to run when I tested it.  And this is only
called when a a/v mime is possibly needed.

**Comment by quozl** [comment]
0.6 seconds sounds okay.  But what hardware did you test on?  Were the filesystem caches empty or populated?

File: src/sugar3/mime.py [hunk 1]
```diff
@@ -94,6 +93,33 @@ def _get_supported_image_mime_types():
 }]
 
 
+def _set_av_mimes():
+    global _generic_types
```
Comment (SAMdroid-apps) [inline src/sugar3/mime.py]: Oops, I don't need this global, I will re push

**Comment by SAMdroid-apps** [comment]
Ok, so I did some more tests.  On an XO4 with on no file system caches, the test took 1.442s.  On my laptop (Core 2 Duo Mobile @ 2GHz) with file caches cleared, it took 1.27s at 50% cpu.  What do you think?

**Comment by SAMdroid-apps** [comment]
Replacing for some caching :musical_note:

**Pull request closed without merging**